File: code/cpp/lambda.cpp
#define unused __attribute ((unused))

#include <vector>
#include <algorithm>

using namespace std;

int main(unused int argc, unused char **argv) {
// BEGIN_CODE
    vector<int> vals = {-1, -2, -3, 1, 2, 3, 0};
    int p = 2;

    sort(vals.begin(),
         vals.end(),
         [p](int a, int b) {
            return pow(a, p) < pow(b, p);
            });

// END_CODE

    return 0;
}

File: code/java/Lambda.java
import java.lang.Math;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Lambda {
    public static void main(String[] args) {
// BEGIN_CODE
    List<Integer> vals = Arrays.asList(-1, -2, -3, 0, 1, 2, 3);
    int p = 2;

    Collections.sort(vals, (a, b) -> 
            Integer.compare(Math.pow(a, p), Math.pow(b, p)));

// END_CODE
    }
}
